### PR TITLE
Interest on user balances

### PR DESCRIPTION
**Pull request by kduffie**

### CODE REVIEW

File: public/src/hash-router.js [hunk 1]
```diff
@@ -18,7 +18,9 @@ class HashRouter extends Polymer.Element {
   }
 
   onRoute() {
-    this.dispatchEvent(new CustomEvent('route-change', { bubbles: true, composed: true, detail: { route: this.route } }));
+    setTimeout(() => {
+      this.dispatchEvent(new CustomEvent('route-change', { bubbles: true, composed: true, detail: { route: this.route } }));
+    }, 100);
   }
```
Comment (pshihn) [inline public/src/hash-router.js]: Why do you need this delay?

File: public/src/app-bar.html [hunk 2]
```diff
@@ -115,8 +115,17 @@
         });
       }
 
-      _refreshBalance(reg) {
-        this.balance = "ℂ" + (reg.status.userBalance || 0).toFixed(3);
+      _refreshBalance(reg, fetched) {
+        setInterval(() => {
+          this.balance = "ℂ" + this._truncateBalance($core.balance);
+        }, 1000);
+      }
```
Comment (pshihn) [inline public/src/app-bar.html]: no need for delay here

File: public/src/app-bar.html [hunk 3]
```diff
@@ -115,8 +115,17 @@
...
+
+      _truncateBalance(value) {
+        if (!value) {
+          return 0;
+        }
+        return (Math.floor(value * 1000) / 1000).toFixed(3);
       }
```
Comment (pshihn) [inline public/src/app-bar.html]: This will always produce .000 
You may want to remove the floor call.

Comment (kduffie) [reply public/src/app-bar.html]: 10.123456 * 1000 = 101234.56
Math.floor(101234.56) = 10123
10123 / 1000 = 10.123
10.123.toFixed(3) = 10.123

Am I missing something?

File: public/src/app-bar.html (hunk 2 above)
Comment (kduffie) [reply public/src/app-bar.html]: This is setInterval, not setTimeout.  But you're right.  Should set it first, and then repeatedly after that.

File: public/src/hash-router.js (hunk 1 above)
Comment (kduffie) [reply public/src/hash-router.js]: Leftover from debugging.  Thx.

File: public/src/app-bar.html (hunk 3 above)
Comment (pshihn) [reply public/src/app-bar.html]: I misread the parenthesis. all good.

**Pull request merged**